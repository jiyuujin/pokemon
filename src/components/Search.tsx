import React from 'react'
import { NekoInput } from 'nekohack-ui'

type Props = {
    text: string
    setText: Function
}

export const Search = (props: Props) => {
    return (
        <div className="search__item">
            <label>{props.text}</label>
            <NekoInput
                value={props.text}
                placeholder="検索してください"
                onChange={props.setText}
            />
        </div>
    )
}
