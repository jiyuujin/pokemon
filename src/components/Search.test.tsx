import React from 'react'
import renderer from 'react-test-renderer'

import { Search } from './Search'

test('Render component', () => {
    const component = renderer.create(
        <Search text={'unown'} setText={() => {}} />
    )
    expect(component).toMatchSnapshot()
})
