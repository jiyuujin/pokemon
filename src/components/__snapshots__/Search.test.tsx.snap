// Vitest Snapshot v1

exports[`Render component 1`] = `
<div
  className="search__item"
>
  <label>
    unown
  </label>
  <input
    className="sc-gtsrHT akHhg"
    onChange={[Function]}
    placeholder="検索してください"
    type="text"
    value="unown"
  />
</div>
`;
